docker_ci_deploy/__main__.py

```diff
@@ -26,41 +27,104 @@ def _strip_image_registry(image):
     return image
 
 
-def _docker_cmd(*cmds, **kwargs):
-    """ Run a Docker command or print it if ``dry_run=True``. """
-    if kwargs.get('dry_run', False):
-        if 'obfuscated' in kwargs:
-            cmds = kwargs['obfuscated']
-        print(' '.join(('docker',) + cmds))
-    else:
-        subprocess.check_output(('docker',) + cmds)
-
-
-def docker_tag(in_tag, out_tag, **kwargs):
-    """ Run ``docker tag`` with the given tags. """
-    _docker_cmd('tag', in_tag, out_tag, **kwargs)
-
-
-def docker_login(username, password, registry, **kwargs):
-    """ Run ``docker login`` with the given credentials. """
-    cmd = [
-        'login',
-        '--username', username,
-        '--password', password,
-        registry,
-    ]
-    obfuscated = list(cmd)
-    obfuscated[4] = '<password>'
-    kwargs['obfuscated'] = tuple(obfuscated)
-    _docker_cmd(*cmd, **kwargs)
-
-
-def docker_push(tag, **kwargs):
-    """ Run ``docker push`` with the given tag. """
-    _docker_cmd('push', tag, **kwargs)
+class DockerCiDeployRunner(object):
+
+    def __init__(self, executable='docker', dry_run=False, verbose=False):
+        self.executable = executable
+        self.dry_run = dry_run
+        self.verbose = verbose
+
+    def _log(self, *args, **kwargs):
+        if kwargs.get('if_verbose', False) and not self.verbose:
+            return
+        print(*args, **kwargs)
```

Comment:
Does `print()` accept keyword args? Do we want to use `kwargs.pop('if_verbose', False)` to remove our own keyword arg before passing it on?